{
  "info": {
    "author": "A. Random Developer",
    "author_email": "author@example.com",
    "bugtrack_url": null,
    "classifiers": [
      "Development Status :: 3 - Alpha",
      "Intended Audience :: Developers",
      "License :: OSI Approved :: MIT License",
      "Programming Language :: Python :: 3",
      "Programming Language :: Python :: 3 :: Only",
      "Programming Language :: Python :: 3.10",
      "Programming Language :: Python :: 3.7",
      "Programming Language :: Python :: 3.8",
      "Programming Language :: Python :: 3.9",
      "Topic :: Software Development :: Build Tools"
    ],
    "description": "# FastAPI Fancy Exceptions\n\n#### Sample code\n```python\nimport uvicorn\nfrom fastapi import FastAPI\nfrom starlette import status\nfrom starlette.requests import Request\nfrom starlette.responses import JSONResponse\nfrom src.fastapi_fancy_exceptions import FancyHTTPException\n\n\napp = FastAPI()\n\n\nclass AuthException(FancyHTTPException):\n    code = \"auth_error\"\n    type = \"AuthError\"\n    message = \"Auth error\"\n    status_code = status.HTTP_401_UNAUTHORIZED\n\n\n@app.exception_handler(FancyHTTPException)\nasync def http_exception_accept_handler(request: Request, exc: FancyHTTPException) -> JSONResponse:\n    return JSONResponse(\n        status_code=exc.status_code,\n        content=[{\"code\": exc.code, \"type\": exc.type, \"message\": exc.message}]\n    )\n\n\n@app.get(\"/\")\nasync def root():\n    raise AuthException()\n\nuvicorn.run(app, host=\"localhost\", port=8000)\n```\n\n#### This code will lead to this response with status code 401:\n\n```json\n[\n  {\n    \"code\": \"auth_error\",\n    \"type\": \"AuthError\",\n    \"message\": \"Auth error\"\n  }\n]\n```\n",
    "description_content_type": "text/markdown",
    "docs_url": null,
    "download_url": "",
    "downloads": {
      "last_day": -1,
      "last_month": -1,
      "last_week": -1
    },
    "home_page": "https://github.com/pypa/sampleproject",
    "keywords": "sample,setuptools,development",
    "license": "",
    "maintainer": "",
    "maintainer_email": "",
    "name": "fastapi-fancy-exceptions",
    "package_url": "https://pypi.org/project/fastapi-fancy-exceptions/",
    "platform": null,
    "project_url": "https://pypi.org/project/fastapi-fancy-exceptions/",
    "project_urls": {
      "Bug Reports": "https://github.com/pypa/sampleproject/issues",
      "Funding": "https://donate.pypi.org",
      "Homepage": "https://github.com/pypa/sampleproject",
      "Say Thanks!": "http://saythanks.io/to/example",
      "Source": "https://github.com/pypa/sampleproject/"
    },
    "release_url": "https://pypi.org/project/fastapi-fancy-exceptions/0.0.3/",
    "requires_dist": [
      "check-manifest ; extra == 'dev'",
      "coverage ; extra == 'test'"
    ],
    "requires_python": ">=3.7, <4",
    "summary": "A sample Python project",
    "version": "0.0.3",
    "yanked": false,
    "yanked_reason": null
  },
  "last_serial": 15281465,
  "releases": {
    "0.0.2": [
      {
        "comment_text": "",
        "digests": {
          "blake2b_256": "d3afb114dc203ce44cf052d724e255270bda035ec4ea8519c47295d298bc8fcf",
          "md5": "3d0a4f879dc1ceae782fcedb49924c71",
          "sha256": "988f9fb77f0ff60c80d3ad841a5ce75d298bc72729a1478ee90b3e27a8f40999"
        },
        "downloads": -1,
        "filename": "fastapi_fancy_exceptions-0.0.2-py3-none-any.whl",
        "has_sig": false,
        "md5_digest": "3d0a4f879dc1ceae782fcedb49924c71",
        "packagetype": "bdist_wheel",
        "python_version": "py3",
        "requires_python": ">=3.7, <4",
        "size": 4930,
        "upload_time": "2022-10-02T13:16:11",
        "upload_time_iso_8601": "2022-10-02T13:16:11.287452Z",
        "url": "https://files.pythonhosted.org/packages/d3/af/b114dc203ce44cf052d724e255270bda035ec4ea8519c47295d298bc8fcf/fastapi_fancy_exceptions-0.0.2-py3-none-any.whl",
        "yanked": false,
        "yanked_reason": null
      },
      {
        "comment_text": "",
        "digests": {
          "blake2b_256": "f2e1817a5a300f1bf7e6f1f0ff270bc5bc148f1ebe866a83f611e76954615eb1",
          "md5": "e9025fcb37c066991e125b924926f8b6",
          "sha256": "df2c0af2603d990d118c2eee47973a4410c8ac04e44049ec673a59b0e9b8f546"
        },
        "downloads": -1,
        "filename": "fastapi-fancy-exceptions-0.0.2.tar.gz",
        "has_sig": false,
        "md5_digest": "e9025fcb37c066991e125b924926f8b6",
        "packagetype": "sdist",
        "python_version": "source",
        "requires_python": ">=3.7, <4",
        "size": 6738,
        "upload_time": "2022-10-02T13:16:15",
        "upload_time_iso_8601": "2022-10-02T13:16:15.367963Z",
        "url": "https://files.pythonhosted.org/packages/f2/e1/817a5a300f1bf7e6f1f0ff270bc5bc148f1ebe866a83f611e76954615eb1/fastapi-fancy-exceptions-0.0.2.tar.gz",
        "yanked": false,
        "yanked_reason": null
      }
    ],
    "0.0.3": [
      {
        "comment_text": "",
        "digests": {
          "blake2b_256": "79f160815c61348d9a450744763e47f0e720db6526c258a3ddfcd9ce2063a83b",
          "md5": "108caee2bcb59f6719486e7edde7ff8f",
          "sha256": "893ccfd6d9e7c0d8a9f176255c77d7e3f872fb9834583d44cb217fb3bc6bbbcd"
        },
        "downloads": -1,
        "filename": "fastapi_fancy_exceptions-0.0.3-py3-none-any.whl",
        "has_sig": false,
        "md5_digest": "108caee2bcb59f6719486e7edde7ff8f",
        "packagetype": "bdist_wheel",
        "python_version": "py3",
        "requires_python": ">=3.7, <4",
        "size": 4319,
        "upload_time": "2022-10-02T13:16:13",
        "upload_time_iso_8601": "2022-10-02T13:16:13.350790Z",
        "url": "https://files.pythonhosted.org/packages/79/f1/60815c61348d9a450744763e47f0e720db6526c258a3ddfcd9ce2063a83b/fastapi_fancy_exceptions-0.0.3-py3-none-any.whl",
        "yanked": false,
        "yanked_reason": null
      },
      {
        "comment_text": "",
        "digests": {
          "blake2b_256": "6e01d596c34826c10ba8fdbf38c4b3ab0b25f6f439d0a985047d4d6ebb30b842",
          "md5": "50efc88fdd7e3d90be6e41ab4fb0cf3d",
          "sha256": "2a60dcc0b14799532a29d669269aace2537910ac7b7bf786c654fb7a2c793bc0"
        },
        "downloads": -1,
        "filename": "fastapi-fancy-exceptions-0.0.3.tar.gz",
        "has_sig": false,
        "md5_digest": "50efc88fdd7e3d90be6e41ab4fb0cf3d",
        "packagetype": "sdist",
        "python_version": "source",
        "requires_python": ">=3.7, <4",
        "size": 6446,
        "upload_time": "2022-10-02T13:16:16",
        "upload_time_iso_8601": "2022-10-02T13:16:16.823081Z",
        "url": "https://files.pythonhosted.org/packages/6e/01/d596c34826c10ba8fdbf38c4b3ab0b25f6f439d0a985047d4d6ebb30b842/fastapi-fancy-exceptions-0.0.3.tar.gz",
        "yanked": false,
        "yanked_reason": null
      }
    ]
  },
  "urls": [
    {
      "comment_text": "",
      "digests": {
        "blake2b_256": "79f160815c61348d9a450744763e47f0e720db6526c258a3ddfcd9ce2063a83b",
        "md5": "108caee2bcb59f6719486e7edde7ff8f",
        "sha256": "893ccfd6d9e7c0d8a9f176255c77d7e3f872fb9834583d44cb217fb3bc6bbbcd"
      },
      "downloads": -1,
      "filename": "fastapi_fancy_exceptions-0.0.3-py3-none-any.whl",
      "has_sig": false,
      "md5_digest": "108caee2bcb59f6719486e7edde7ff8f",
      "packagetype": "bdist_wheel",
      "python_version": "py3",
      "requires_python": ">=3.7, <4",
      "size": 4319,
      "upload_time": "2022-10-02T13:16:13",
      "upload_time_iso_8601": "2022-10-02T13:16:13.350790Z",
      "url": "https://files.pythonhosted.org/packages/79/f1/60815c61348d9a450744763e47f0e720db6526c258a3ddfcd9ce2063a83b/fastapi_fancy_exceptions-0.0.3-py3-none-any.whl",
      "yanked": false,
      "yanked_reason": null
    },
    {
      "comment_text": "",
      "digests": {
        "blake2b_256": "6e01d596c34826c10ba8fdbf38c4b3ab0b25f6f439d0a985047d4d6ebb30b842",
        "md5": "50efc88fdd7e3d90be6e41ab4fb0cf3d",
        "sha256": "2a60dcc0b14799532a29d669269aace2537910ac7b7bf786c654fb7a2c793bc0"
      },
      "downloads": -1,
      "filename": "fastapi-fancy-exceptions-0.0.3.tar.gz",
      "has_sig": false,
      "md5_digest": "50efc88fdd7e3d90be6e41ab4fb0cf3d",
      "packagetype": "sdist",
      "python_version": "source",
      "requires_python": ">=3.7, <4",
      "size": 6446,
      "upload_time": "2022-10-02T13:16:16",
      "upload_time_iso_8601": "2022-10-02T13:16:16.823081Z",
      "url": "https://files.pythonhosted.org/packages/6e/01/d596c34826c10ba8fdbf38c4b3ab0b25f6f439d0a985047d4d6ebb30b842/fastapi-fancy-exceptions-0.0.3.tar.gz",
      "yanked": false,
      "yanked_reason": null
    }
  ],
  "vulnerabilities": []
}